src/app/components/utility/userStateService.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Created by chima on 11/19/15.
+ */
+'use strict'
+
+angular.module('utility')
+  .service('userStateService', function (ehaCouchDbAuthService, locationService, $localForage, $q) {
+    var self = this
+    self.stateMap = {states: [], selectedState: ''}
+
+    function initializeStateVariables () {
+      $localForage.clear()
+      self.stateMap.states = []
+      self.stateMap.selectedState = ''
+    }
+
+    this.getUserSelectedState = function (byId) {
+      return ehaCouchDbAuthService.getCurrentUser()
+        .then(function (user) {
+          if (!user.userCtx.name) {
+            return ''
+          }
+          return $localForage.getItem(user.userCtx.name)
```

Comment:
`$localForage` isn't a top-level dependency (it's a transitive dependency from eha.couchdb-auth). Good idea to store the user's previously selected state though (from a UX point of view). localforage will need adding as a top-level dependency (in `bower.json`). (Later, I'd be tempted to replace this with [PouchDB local docs](http://pouchdb.com/guides/local-documents.html), though there's little point until pluggable local storage wrapper support is added to eha.couchdb-auth).